src/transaction/log_recovery_redo_perf.hpp

```diff
@@ -0,0 +1,136 @@
+				"Counter redo_or_push_prep", "Timer redo_or_push_prep"),
+      cubperf::stat_definition (PERF_STAT_ID_REDO_OR_PUSH_DO_SYNC, cubperf::stat_definition::COUNTER_AND_TIMER,
+				"Counter redo_or_push_do_sync", "Timer redo_or_push_do_sync"),
+      cubperf::stat_definition (PERF_STAT_ID_REDO_OR_PUSH_DO_ASYNC, cubperf::stat_definition::COUNTER_AND_TIMER,
+				"Counter redo_or_push_do_async", "Timer redo_or_push_do_async"),
+      cubperf::stat_definition (PERF_STAT_ID_COMMIT_ABORT, cubperf::stat_definition::COUNTER_AND_TIMER,
+				"Counter commit_abort", "Timer commit_abort"),
+      cubperf::stat_definition (PERF_STAT_ID_WAIT_FOR_PARALLEL, cubperf::stat_definition::COUNTER_AND_TIMER,
+				"Counter wait_for_parallel", "Timer wait_for_parallel"),
+      cubperf::stat_definition (PERF_STAT_ID_FINALIZE, cubperf::stat_definition::COUNTER_AND_TIMER,
+				"Counter finalize", "Timer finalize"),
+    }
+    , m_values { nullptr }
+    {
+      if ( (pstat_Global.activation_flag & PERFMON_ACTIVATION_FLAG_LOG_RECOVERY_REDO)
+	   == PERFMON_ACTIVATION_FLAG_LOG_RECOVERY_REDO )
+	{
+	  m_values = m_definition.create_statset ();
+	}
+    }
+
+    log_recovery_redo_perf_stat (const log_recovery_redo_perf_stat &) = delete;
+    log_recovery_redo_perf_stat (log_recovery_redo_perf_stat &&) = delete;
+
+    inline ~log_recovery_redo_perf_stat ()
+    {
+      if (m_values != nullptr)
+	{
+	  delete m_values;
+	  m_values = nullptr;
+	}
+    }
+
+    log_recovery_redo_perf_stat &operator = (const log_recovery_redo_perf_stat &) = delete;
+    log_recovery_redo_perf_stat &operator = (log_recovery_redo_perf_stat &&) = delete;
+
+  public:
+    inline void time_and_increment (cubperf::stat_id a_stat_id) const
+    {
+      if (m_values != nullptr)
+	{
+	  m_definition.time_and_increment (*m_values, a_stat_id);
+	}
+    }
+
+    inline void log () const
+    {
+      if (m_values != nullptr)
+	{
+	  std::vector < cubperf::stat_value > perf_stat_results;
+	  perf_stat_results.resize (m_definition.get_value_count (), 0LL);
+	  m_definition.get_stat_values_with_converted_timers<std::chrono::microseconds> (
+		  *m_values, perf_stat_results.data ());
+
+	  std::stringstream perf_stat_ss;
+	  perf_stat_ss << "Log Recovery Redo statistics:" << std::endl;
+	  for (std::size_t perf_stat_idx = 0
+					   ; perf_stat_idx < m_definition.get_value_count (); ++perf_stat_idx)
+	    {
+	      perf_stat_ss.imbue (std::locale (""));
+	      perf_stat_ss << '\t' << m_definition.get_value_name (perf_stat_idx)
+			   << " μs: " << perf_stat_results[perf_stat_idx] << std::endl;
+	    }
+	  const std::string perf_stat_str = perf_stat_ss.str ();
+	  _er_log_debug (ARG_FILE_LINE, perf_stat_str.c_str ());
+	}
+    }
+
+  private:
+    const cubperf::statset_definition m_definition;
+    cubperf::statset *m_values;
+};
+
+extern log_recovery_redo_perf_stat *log_recovery_redo_perf_stat_ptr;
```

Comment:
Remove the global, use a local variable in log_recovery_redo and pass as argument to log_rv_redo_record_sync_or_dispatch_async.

One of the following tasks will have to be to group some (most?) of the log_rv_redo_record_sync_or_dispatch_async arguments into a class.